### PR TITLE
VideoPlayer: refactor DemuxerClient (ex PVR)

### PR DESCRIPTION
**Pull request by FernetMenta**
see title,
This makes former DVDDemuxPVRClient generic, DVDDemuxClient. The PVR specific methods were moved to the new IDemux interface that is implemented by DVDInputStreamPVRManager.

This has two major advantages:
- centralize PVR specific methods that were spreaded over demuxer and input stream
- give other components and addon the change to implement IDemux and implement their own demuxer

### CODE REVIEW

**Comment by Hedda** [comment]
Sorry of bikeshedding but thought that you had planned to remove the "DVD*" prefix now with the implementation of the new "VideoPlayer" concept? Might you not just rename DVDDemuxClient.* / DVDDemuxPacket.* to VideoPlayerDemuxClient.* / VideoPlayerDemuxPacket.* right away?

Understand that the "DVD*" prefix used in DVDPlayer and its source code files are only named so for historical reasons when the DVDPlayer was first specifically made only to serve as a DVD-Video player core with DVD-menu support, and the new VideoPlayer should not keep the DVDPlayer name?

**Comment by FernetMenta** [comment]
I intend to do the rename when we have cmake and got rid of those nasty project files for xcode and vs.

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+  int len = av_parser_parse2(stream->m_parser,
+                             stream->m_context, &outbuf, &outbuf_size,
+                             pkt->pData, pkt->iSize,
+                             (int64_t)(pkt->pts * DVD_TIME_BASE),
+                             (int64_t)(pkt->dts * DVD_TIME_BASE),
+                             0);
+  // our parse is setup to parse complete frames, so we don't care about outbufs
+  if(len >= 0)
+  {
+#define CHECK_UPDATE(st, trg, src, invalid) do { \
```
Comment (Jalle19) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: Any chance you could refactor this macro into a method while you're at it?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: I thought about this already because I hate macros :)

Comment (Jalle19) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: Yeah I know ;-) I remember having to debug part of this code once and it was almost impossible due to this macro.

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
OSX build error unrelated

**Comment by fritsch** [comment]
@FernetMenta people on forum complain that this broken tvheadend - I did not see a logfile yet - but be prepared.

**Pull request merged**